src/verify_cert.rs

```diff
@@ -22,12 +22,20 @@ use crate::crl::RevocationOptions;
 use crate::der::{self, FromDer};
 use crate::{signed_data, subject_name, Error};
 
-pub(crate) struct ChainOptions<'a> {
-    pub(crate) eku: KeyUsage,
-    pub(crate) supported_sig_algs: &'a [&'a dyn SignatureVerificationAlgorithm],
-    pub(crate) trust_anchors: &'a [TrustAnchor<'a>],
-    pub(crate) intermediate_certs: &'a [CertificateDer<'a>],
-    pub(crate) revocation: Option<RevocationOptions<'a>>,
+/// Input context for verifying a certificate chain for a particular purpose.
+pub struct ChainOptions<'a> {
```

Comment:
Having some second thoughts about whether this is a useful transformation. Most of all, as @cpu pointed out, the intermediates probably don't belong in here. But also, while I dislike the long argument list on `verify_for_usage()`, I'm also not completely convinced this is an improvement.

Any thoughts?